### ISSUE TITLE
TransactionRequiredException in Grails 4

### ISSUE THREAD

**Issue by verglor** [kefirfromperm/grails-asynchronous-mail#91]
It seems there are missing transaction declarations - see [GORM manual section 1.2.5](http://gorm.grails.org/latest/hibernate/manual/index.html#_transactions_now_required_for_all_operations)

**Comment by verglor** [comment]
[Workaround](https://github.com/grails/grails-core/issues/11376#issuecomment-513561668) is to configure `hibernate.allow_update_outside_transaction = true`

**Comment by xpusostomos** [comment]
I think this is the wrong solution. After grails 3.1 services aren't transactional by default.  The AsynchronousMailProcessService isn't marked as @Transactional. The correct solution is to mark the service as transactional. Doing global hibernate workarounds is bad because plugins shouldn't assume a particular global setup.

**Comment by verglor** [comment]
@xpusostomos you are right, it's just a quick workaround. That's why this issue is still open and unresolved.
